front/src/applications/stdcm/hooks/useConflicts.ts

```diff
@@ -0,0 +1,108 @@
+
+const useConflictsMessages = (
+  t: TFunction,
+  conflictData?: Conflict[],
+  pathProperties?: ManageTrainSchedulePathProperties
+) => {
+  const [trackConflicts, setTrackConflicts] = useState<string[]>([]);
+  const [workConflicts, setWorkConflicts] = useState<string[]>([]);
+
+  useEffect(() => {
+    if (!pathProperties || !conflictData) return;
+
+    const generateConflictMessages = () => {
+      const processedConflictsData = processConflicts(conflictData, pathProperties);
+
+      const trackConflictsData = processedConflictsData.filter(
+        (conflict) => conflict.conflictType === 'Spacing'
+      );
+
+      const workConflictsData = processedConflictsData.filter(
+        (conflict) => conflict.conflictType !== 'Spacing'
+      );
+
+      const trackMessages: string[] = [];
+      trackConflictsData.slice(0, 2).forEach((conflict) => {
+        const { waypointBefore, waypointAfter, startDate, endDate, startTime, endTime } = conflict;
+
+        if (startDate === endDate) {
+          trackMessages.push(
+            t('trackConflictSameDay', {
+              waypointBefore,
+              waypointAfter,
+              startTime,
+              endTime,
+              startDate,
+            })
+          );
+        } else {
+          trackMessages.push(
+            t('trackConflict', {
+              waypointBefore,
+              waypointAfter,
+              startDate,
+              endDate,
+              startTime,
+              endTime,
+            })
+          );
+        }
+      });
+
+      const remainingTrackConflicts = trackConflictsData.length - 2;
+      if (remainingTrackConflicts > 0) {
+        trackMessages.push(t('remainingTrackConflicts', { remainingTrackConflicts }));
+      }
+
+      const workMessages: string[] = [];
+      workConflictsData.slice(0, 2).forEach((conflict) => {
+        const { waypointBefore, waypointAfter, startDate, endDate, startTime, endTime } = conflict;
+
+        if (startDate === endDate) {
+          workMessages.push(
+            t('workConflictSameDay', {
+              waypointBefore,
+              waypointAfter,
+              startDate,
+              startTime,
+              endTime,
+            })
+          );
+        } else {
+          workMessages.push(
+            t('workConflict', {
+              waypointBefore,
+              waypointAfter,
+              startDate,
+              startTime,
+              endDate,
+              endTime,
+            })
+          );
+        }
+      });
+
+      const remainingWorkConflicts = workConflictsData.length - 2;
+      if (remainingWorkConflicts > 0) {
+        workMessages.push(t('remainingWorkConflicts', { remainingWorkConflicts }));
+      }
+
+      // Update the state with generated messages
+      setTrackConflicts(trackMessages);
+      setWorkConflicts(workMessages);
+    };
+
+    generateConflictMessages();
+  }, [t, pathProperties, conflictData]);
```

Comment:
After the function parameter change

```suggestion
  }, [t, selectedSimulationOutput]);
```